From dda86b678a2c3260c01d616e287e728504d50036 Mon Sep 17 00:00:00 2001
From: Tomas Glozar <tglozar@gmail.com>
Date: Sat, 7 Dec 2024 18:26:31 +0100
Subject: [PATCH 88/94] ia64: Re-introduce efi_initialize_iomem_resources

Commit 21b91d40575f ("efi: Remove unused declaration
efi_initialize_iomem_resources()") removed
efi_initialize_iomem_resources which was ia64-specific at that point.

Bring back the declaration for ia64 in arch/ia64/include/asm/efi.h.

Signed-off-by: Tomas Glozar <tglozar@gmail.com>
---
 arch/ia64/include/asm/efi.h | 2 ++
 arch/ia64/kernel/setup.c    | 1 +
 2 files changed, 3 insertions(+)

diff --git a/arch/ia64/include/asm/efi.h b/arch/ia64/include/asm/efi.h
index 6a4a50d8f19a..7baf1aa93d45 100644
--- a/arch/ia64/include/asm/efi.h
+++ b/arch/ia64/include/asm/efi.h
@@ -9,5 +9,7 @@ void efi_map_pal_code(void);
 void efi_memmap_walk(efi_freemem_callback_t, void *);
 void efi_memmap_walk_uc(efi_freemem_callback_t, void *);
 void efi_gettimeofday(struct timespec64 *ts);
+void efi_initialize_iomem_resources(struct resource *code_resource,
+		struct resource *data_resource, struct resource *bss_resource);
 
 #endif
diff --git a/arch/ia64/kernel/setup.c b/arch/ia64/kernel/setup.c
index 8ddf263e1a40..8babc91bfb45 100644
--- a/arch/ia64/kernel/setup.c
+++ b/arch/ia64/kernel/setup.c
@@ -53,6 +53,7 @@
 #include <linux/kexec.h>
 #include <linux/crash_dump.h>
 
+#include <asm/efi.h>
 #include <asm/mca.h>
 #include <asm/meminit.h>
 #include <asm/page.h>
-- 
2.25.1

